### PR TITLE
Add indexing functionality to view results one by one

### PR DESCRIPTION
**Pull request by Ponmo**
#### Description:
- Added a number input field to view certain results one-by-one.

#### Testing:
- Tested the input field on multiple different states including pending, when a new query is ran, when the input field is focused, etc. and seemed to work appropriately.
- npm run test did not find any new errors.

#### Relevant Issue:
https://boi-ucsb.atlassian.net/browse/ENG-230?atlOrigin=eyJpIjoiODc2MTE1ZDkwMjM0NDYzZTllNmZiNmZjYTMwZWQ0Y2QiLCJwIjoiaiJ9

Screenshot 2024-10-14 at 1 17 12 PM

### CODE REVIEW

File: src/app/map/map.component.ts [hunk 1]
```diff
@@ -185,14 +202,15 @@ export class MapComponent implements AfterViewInit {
...
-      data: this._features,
+      data: this._page === 0 ? this._features : [this._features[this._page - 1]],
       pickable: true,
       autoHighlight: true,
       highlightColor: [219, 68, 55], // #DB4437
       stroked: this.hasStroke(),
       filled: true,
       extruded: false,
       elevationScale: 0,
+      binary: true,
```
Comment (crlang44) [inline src/app/map/map.component.ts]: What does this do?

**Review by crlang44 — APPROVED** [review]
Awesome job! It works great! I had one comment for the binary attribute but it seems to work either way.

File: src/app/map/map.component.ts (hunk 1 above)
Comment (Ponmo) [reply src/app/map/map.component.ts]: Oh, good catch! I was just playing around with some of the functionality in the maps API. I will publish a new revision.

File: src/app/main/main.component.ts [hunk 2]
```diff
@@ -505,6 +514,51 @@ ${USER_QUERY_END_MARKER}\n
...
+        this.page = this.rows.length;
+      }
+      else {
+        this.page = newPage;
+      }
+    }
+    this.isEditingPagination = false;
+  }
+
+
```
Comment (crlang44) [inline src/app/main/main.component.ts:558]: Remove extra whitespace

File: src/app/map/map.component.ts [hunk 3]
```diff
@@ -53,6 +53,7 @@ export class MapComponent implements AfterViewInit {
   private _styles: StyleRule[] = [];
   private _geoColumn: string;
   private _activeGeometryTypes = new Set<string>();
+  private _geoJSONLayer = new GeoJsonLayer();
```
Comment (crlang44) [inline src/app/map/map.component.ts:56]: This variable is not used, can remove.

File: src/app/main/main.component.ts [hunk 4]
```diff
@@ -505,6 +514,51 @@ ${USER_QUERY_END_MARKER}\n
...
+    this.isEditingPagination = true;
+    setTimeout(() => {
+      this.pageInput.nativeElement.focus();
+    });
+  }
+
+  finishPaginationEditing(event: Event) {
+    const input = event.target as HTMLInputElement;
+    const value = input.value;
+
```
Comment (crlang44) [inline src/app/main/main.component.ts:538]: Remove extra line

File: src/app/main/main.component.ts [hunk 5]
```diff
@@ -505,6 +514,51 @@ ${USER_QUERY_END_MARKER}\n
 
   }
 
+  paginate(_page: number) {
+    // Left button was pressed
+    console.log(this.page, this.totalRows)
```
Comment (crlang44) [inline src/app/main/main.component.ts:519]: Remove console.log

**Pull request merged**